GCRCatalogs/dc2_coadd.py

```diff
@@ -12,16 +12,66 @@
 
 __all__ = ['DC2CoaddCatalog']
 
+FILE_PATTERN = r'merged_tract_\d+\.hdf5'
+GROUP_PATTERN = r'coadd_\d+_\d\d$'
+
+
+def calc_cov(ixx_err, iyy_err, ixy_err):
+    """Calculate the covariance between three arrays of second moments
+
+    Args:
+        ixx_err (float): The error in the second moment Ixx
+        iyy_err (float): The error in the second moment Iyy
+        ixy_err (float): The error in the second moment Ixy
+
+    Returns:
+        The covariance between the given arrays
+    """
+    # If need be, this can be generalized using a mesh grid
+    out_data = np.array([
+        ixx_err * ixx_err,
+        ixx_err * iyy_err,
+        ixx_err * ixy_err,
+        iyy_err * iyy_err,
+        iyy_err * ixy_err,
+        ixy_err * ixy_err])
+
+    return out_data.transpose()
+
+
+def create_basic_flag_mash(*flags):
+    """
+    generate a mask for a set of flags
+    for each item, mask will be true if and only if all flags are false
+    """
+    out = np.ones(len(flags[0]), np.bool)
+    for flag in flags:
+        out &= (~flag)
+    return out
+
+
 class DC2CoaddCatalog(BaseGenericCatalog):
```

Comment:
For this PR we should keep this, but I agree that if we start using HSC significantly it will make sense to make a more generic class that we inherit from.